unlock-app/src/components/interface/checkout/main/useHooks.ts

```diff
@@ -0,0 +1,42 @@
+import { useQuery } from '@tanstack/react-query'
+import networks from '@unlock-protocol/networks'
+import { HookType } from '@unlock-protocol/types'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+
+interface PasswordHookSignersProps {
+  lockAddress: string
+  network: number
+}
+export function usePasswordHookSigners({
+  lockAddress,
+  network,
+}: PasswordHookSignersProps) {
+  const web3Service = useWeb3Service()
+  const { getWalletService } = useAuth()
+  // get password hook contract by network
```

Comment:
Actually let's get it from the lock contract `lock.onPurchaseHook`  this will be more robust and makes more sense because we check the hook contract from the lock itself.